### ISSUE TITLE
Cannot parse @

### ISSUE THREAD

**Issue by Jaxing** [sqlfluff/sqlfluff#2871]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I tried to lint some queries that I run in bigquery using [query parameters](https://cloud.google.com/bigquery/docs/parameterized-queries#python). A query parameter is prefixed with an @, however I get this output `L:  48 | P:  47 |  PRS | Line 48, Position 47: Found unparsable section:`
It seems to only happen when the query parameter is used inside a window query.

### Expected Behaviour

I would like to be able to parse query parameters prefixed with @

### Observed Behaviour

Cannot I cannot parse query parameters prefixed with @

### How to reproduce

SQL file
```sql
select
    id,
    sum(something) over( partition by some_id order by some_date rows BETWEEN @query_parameter PRECEDING AND CURRENT ROW) as some_sum
from some_table
where some_column = @query_parameter2
```

Output: 
```
L:   3 | P:  65 |  PRS | Line 3, Position 65: Found unparsable section: ' rows
                       | BETWEEN @query_parameter PRECEDING...'
All Finished 📜 🎉!
```

### Dialect

bigquery

### Version

sqlfluff, version 0.9.4
Python 3.8.12

### Configuration

sqlfluff config
```
[sqlfluff]
templater = jinja
dialect = bigquery

[sqlfluff:indentation]
indented_joins = false
indented_using_on = true
template_blocks_indent = false

[sqlfluff:templater]
unwrap_wrapped_queries = true

[sqlfluff:templater:dbt]
project_dir = dbt-annotell

[sqlfluff:rules]
tab_space_size = 4
indent_unit = space
max_line_length: 140

[sqlfluff:rules:L007]
operator_new_lines = before

[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = lower

[sqlfluff:rules:L030]
# Function names
capitalisation_policy = lower
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Fixed in #2872 which should be included in the release we're planning for later this week.

**Comment by Jaxing** [comment]
Ah great thanks!
